pimcore/models/DataObject/AbstractObject.php

```diff
@@ -533,35 +533,39 @@ public function setLocked($o_locked)
     public function delete()
     {
         \Pimcore::getEventDispatcher()->dispatch(DataObjectEvents::PRE_DELETE, new DataObjectEvent($this));
-
-        // delete childs
-        if ($this->hasChildren([self::OBJECT_TYPE_OBJECT, self::OBJECT_TYPE_FOLDER, self::OBJECT_TYPE_VARIANT])) {
-            // delete also unpublished children
-            $unpublishedStatus = self::doHideUnpublished();
-            self::setHideUnpublished(false);
-            foreach ($this->getChildren([self::OBJECT_TYPE_OBJECT, self::OBJECT_TYPE_FOLDER, self::OBJECT_TYPE_VARIANT], true) as $value) {
-                $value->delete();
+        try {
+            // delete childs
+            if ($this->hasChildren([self::OBJECT_TYPE_OBJECT, self::OBJECT_TYPE_FOLDER, self::OBJECT_TYPE_VARIANT])) {
+                // delete also unpublished children
+                $unpublishedStatus = self::doHideUnpublished();
+                self::setHideUnpublished(false);
+                foreach ($this->getChildren([self::OBJECT_TYPE_OBJECT, self::OBJECT_TYPE_FOLDER, self::OBJECT_TYPE_VARIANT], true) as $value) {
+                    $value->delete();
+                }
+                self::setHideUnpublished($unpublishedStatus);
             }
-            self::setHideUnpublished($unpublishedStatus);
-        }
 
-        // remove dependencies
-        $d = $this->getDependencies();
-        $d->cleanAllForElement($this);
+            // remove dependencies
+            $d = $this->getDependencies();
+            $d->cleanAllForElement($this);
 
-        // remove all properties
-        $this->getDao()->deleteAllProperties();
+            // remove all properties
+            $this->getDao()->deleteAllProperties();
 
-        // remove all permissions
-        $this->getDao()->deleteAllPermissions();
+            // remove all permissions
+            $this->getDao()->deleteAllPermissions();
 
-        $this->getDao()->delete();
+            $this->getDao()->delete();
 
-        // empty object cache
-        $this->clearDependentCache();
+            // empty object cache
+            $this->clearDependentCache();
 
-        //set object to registry
-        \Pimcore\Cache\Runtime::set('object_' . $this->getId(), null);
+            //clear object from registry
+            \Pimcore\Cache\Runtime::set('object_' . $this->getId(), null);
+        } catch (\Exception $e) {
+            \Pimcore::getEventDispatcher()->dispatch(DataObjectEvents::POST_DELETE_FAILURE, new DataObjectEvent($this));
```

Comment:
some goes for here ... we should throw that exception

Reply:
Done ;-)